supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsAsyncJobExecutionActor.scala

```diff
@@ -0,0 +1,255 @@
+package cromwell.backend.impl.aws
+
+import java.nio.file.Path
+
+import akka.actor.{Actor, ActorLogging, ActorRef}
+import better.files.File
+import com.amazonaws.auth.{AWSCredentials, BasicAWSCredentials}
+import com.amazonaws.services.ecs.model._
+import com.amazonaws.services.ecs.{AmazonECSAsync, AmazonECSAsyncClient}
+import cromwell.backend.BackendJobExecutionActor.BackendJobExecutionResponse
+import cromwell.backend.async.AsyncBackendJobExecutionActor.{ExecutionMode, Recover}
+import cromwell.backend.async.{AsyncBackendJobExecutionActor, ExecutionHandle, FailedNonRetryableExecutionHandle, PendingExecutionHandle, SuccessfulExecutionHandle}
+import cromwell.backend.validation.{ContinueOnReturnCode, ContinueOnReturnCodeFlag}
+import cromwell.backend.wdl.OnlyPureFunctions
+import cromwell.backend.{BackendConfigurationDescriptor, BackendJobDescriptor, BackendJobLifecycleActor}
+import cromwell.core.retry.SimpleExponentialBackoff
+import net.ceedubs.ficus.Ficus._
+import wdl4s.expression.WdlFunctions
+import wdl4s.values.WdlValue
+import wdl4s.{Task => _, _}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success, Try}
+
+class AwsAsyncJobExecutionActor(override val jobDescriptor: BackendJobDescriptor,
+                                override val completionPromise: Promise[BackendJobExecutionResponse],
+                                val configurationDescriptor: BackendConfigurationDescriptor,
+                                val serviceRegistryActor: ActorRef)
+  extends Actor with ActorLogging with BackendJobLifecycleActor with AsyncBackendJobExecutionActor {
+
+  // Copypasta
```

Comment:
All of this could be in a trait. I had to draw a line somewhere though, but I believe both the SFS, JES, and now AWS backends could all use the the same code listed under "copypasta".